src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -415,7 +425,7 @@ internal void EnsureThreadRequested()
             }
         }
 
-        internal void MarkThreadRequestSatisfied()
+        internal void MarkThreadRequestSatisfied(bool dequeSuccessful)
```

Comment:
`deque` to `dequeue` OK

Looks like `workItems.IsEmpty()` avoids the freezing segments.

I think this will still improve performance a few more percent.  I will test.
```
internal bool 
ThreadRequestNeeded(int count) => (count < ThreadPoolGlobals.processorCount) &&
            !(workItems.IsEmpty() || (WorkStealingQueueList.wsqActive == 0));
```